File: app/src/main/java/weatherapp/tittojose/me/weatherapp/ui/view/HomeScreen.java
package weatherapp.tittojose.me.weatherapp.ui.view;

import weatherapp.tittojose.me.weatherapp.model.pojo.WeatherModel;

public interface HomeScreen {
    void onWeatherDataLoadSuccess(WeatherModel weatherModel);

    void onWeatherDataLoadFailed();

    void showLoading();

    void hideLoading();

    void onRetryButtonClicked();
}

File: app/src/main/java/weatherapp/tittojose/me/weatherapp/view/HomeScreen.kt
package weatherapp.tittojose.me.weatherapp.view

import weatherapp.tittojose.me.weatherapp.model.pojo.WeatherModel

interface HomeScreen {
	fun onWeatherDataLoadSuccess(weatherModel: WeatherModel)
	fun onWeatherDataLoadFailed()
	fun showLoading()
	fun hideLoading()
	fun onRetryButtonClicked()
}